dawn/renderer/components/ConfigBox.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import {
+  Modal,
+  Button,
+  FormGroup,
+  Form,
+  FormControl,
+  ControlLabel,
+} from 'react-bootstrap';
+
+class UpdateBox extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      ipAddress: this.props.ipAddress,
+    };
+    this.saveChanges = this.saveChanges.bind(this);
+    this.getValidationState = this.getValidationState.bind(this);
+    this.disableUploadUpdate = this.disableUploadUpdate.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+
+  getValidationState() {
+    const current = this.state.ipAddress;
+    let valid = false;
+    const regex = '^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}' +
```

Comment:
This test IP step can be moved into a generic utility file and imported (see other comments)